### PR TITLE
Expose etcd compaction time via environmental variable in GCE

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
Configurable etcd compaction frequency in GCE
```

### CODE REVIEW

File: cluster/common.sh [hunk 1]
```diff
@@ -828,6 +828,11 @@ EOF
     if [ -n "${KUBE_APISERVER_LIVENESS_PROBE_INITIAL_DELAY_SEC:-}" ]; then
       cat >>$file <<EOF
 KUBE_APISERVER_LIVENESS_PROBE_INITIAL_DELAY_SEC: $(yaml-quote ${KUBE_APISERVER_LIVENESS_PROBE_INITIAL_DELAY_SEC})
+EOF
+    fi
+    if [ -n "${ETCD_COMPACTION_INTERVAL_SEC:-}" ]; then
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply cluster/common.sh]: Not necessarily. This isn't needed to make it useful.
I mean, you can set it without doing anything to config-test/default.

[the remark this answers is not in the document]
Comment (shyamjvs) [reply cluster/common.sh]: Yes, I understand. I was asking because it gives more visibility for the option there.
I'm fine either way though.

**Comment by wojtek-t** [comment]
/hold

[I need to verify if that works correctly.]

File: cluster/gce/manifests/kube-apiserver.manifest [hunk 2]
```diff
@@ -167,6 +167,11 @@
   {% set enable_garbage_collector = "--enable-garbage-collector=" + pillar['enable_garbage_collector'] -%}
 {% endif -%}
 
+{% set etcd_compaction_interval = "" %}
```
Comment (porridge) [inline cluster/gce/manifests/kube-apiserver.manifest]: I'd use the `_sec` suffix, since you're appending "s" to the value.

Comment (wojtek-t) [reply cluster/gce/manifests/kube-apiserver.manifest]: Changed partially (to make it compatible with there is for kube_apiserver_request_timeout_sec).
BTW - it seems that we no longer have proper support for those pillars, so we should probably get rid of them completely...

**Comment by wojtek-t** [comment]
OK - I verified that it works in my test cluster.

/unhold

**Pull request merged**